janusgraph-core/src/main/java/org/janusgraph/graphdb/query/graph/GraphCentricQueryBuilder.java

```diff
@@ -239,19 +242,24 @@ public GraphCentricQuery constructQueryWithoutProfile(final ElementCategory resu
         if (limit == 0) return GraphCentricQuery.emptyQuery(resultType);
 
         if (globalConstraints.isEmpty()) {
-            globalConstraints.add(constraints);
+            globalConstraints.add(Collections.singletonList(constraints));
         }
         //Prepare constraints
         final MultiCondition<JanusGraphElement> conditions;
-        if (this.globalConstraints.size() == 1) {
+        if (this.globalConstraints.size() == 1 && this.globalConstraints.get(0).size() == 1) {
```

Comment:
I know that the basic structure of this `if` statement was there before your PR, but I would like to mention that it seems weird to me:
The `constraints` are only added to the `globalConstraints` in the case that `globalConstraints` was empty before. If this is the case, then the `if` branch is taken and `globalConstraints` is never actually used again in the remainder of this method. Therefore `globalConstraints.add(Collections.singletonList(constraints))` is redundant.
In the other case, both `globalConstraints` and `constraints` could possibly co-exist. However, only `globalConstraints` is taken into account when building the query. However, I was not able to construct a test case such that both `globalConstraints` and `constraints` exist at the same time. Is this even possible?

Reply:
This did seem strange to me when I first looked at it, but didn't want to go out of the scope of the issue I was trying to fix so I kept the basic logic intact. If you want, I'll refactor this to remove redundant statements.

Reply:
You don't need to. I just wanted to try to understand the logic behind this.